src/db.c

```diff
@@ -4128,206 +4155,1718 @@ db_speaker_clear_all(void)
   db_query_run("UPDATE speakers SET selected = 0;", 0, 0);
 }
 
+/* Queue */
 
-/* Inotify */
-int
-db_watch_clear(void)
-{
-  return db_query_run("DELETE FROM inotify;", 0, 0);
+static int
+queue_fetch_byitemid(struct db_queue_enum *queue_enum, uint32_t item_id, struct db_queue_item *queue_item, int keep_item);
+
+
+void
+free_queue_item(struct db_queue_item *queue_item, int content_only)
+{
+  if (queue_item->path)
+    free(queue_item->path);
+  if (queue_item->virtual_path)
+    free(queue_item->virtual_path);
+  if (queue_item->title)
+    free(queue_item->title);
+  if (queue_item->artist)
+    free(queue_item->artist);
+  if (queue_item->album_artist)
+    free(queue_item->album_artist);
+  if (queue_item->album)
+    free(queue_item->album);
+  if (queue_item->genre)
+    free(queue_item->genre);
+  if (queue_item->artist_sort)
+    free(queue_item->artist_sort);
+  if (queue_item->album_sort)
+    free(queue_item->album_sort);
+  if (queue_item->album_artist_sort)
+    free(queue_item->album_artist_sort);
+
+  if (!content_only && queue_item)
+    free(queue_item);
 }
 
+/*
+ * Returns the queue version from the admin table
+ *
+ * @return queue version
+ */
 int
-db_watch_add(struct watch_info *wi)
+db_queue_get_version()
```

Comment:
With out this function the caller would have to convert the value to an int. I prefer to keep it. It encapsulates the knowledge how the queue-version is stored in the db.c file.